packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -197,7 +197,11 @@ export interface SpecRestApiProps extends RestApiOptions {
   readonly apiDefinition: ApiDefinition;
 }
 
-abstract class RestApiBase extends Resource implements IRestApi {
+/**
+ * Abstract class that allow for further configuration of a API Gateway RestApi,
+ * such as Resources, Methods, Stages and Deployments.
+ */
+export abstract class MutableRestApi extends Resource implements IRestApi {
```

Comment:
Can we keep this class internal?

Reply:
If you mean, drop the `export` keyword, we cannot.
It is being used in [resource.ts](https://github.com/aws/aws-cdk/pull/8270/files#diff-4c212e4a21caccbdb7f0bb9cbabfb32a).

Reply:
If it is only being used as a base class, jsii should be able to erase it. It just doesn’t need to appear in methods/properties